llvm/lib/SYCLLowerIR/ModuleSplitter.cpp

```diff
@@ -177,6 +177,11 @@ class DependencyGraph {
 
     // We add every function into the graph
     for (const auto &F : M.functions()) {
+
+      if (SplitForRunTimeLinking &&
```

Comment:
The variable name and the behavior are a bit misaligned/nondescriptive to me. It seems like what the bool really controls is it we include SYCL_EXTERNAL functions into the image, maybe we could name the variable `IncludeExternalFunctions` or something

Reply:
Updated variable to ExcludeExternalFunctions.